data.azurerm_management_redis is crashed

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.13.3

### AzureRM Provider Version

4.50.0

### Affected Resource(s)/Data Source(s)

azurerm_managed_redis data source

### Terraform Configuration Files

```hcl
terraform {
  cloud {
    organization = ""
    hostname     = "app.terraform.io"

    workspaces {
      name = ""
    }
  }
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">= 4.50.0"
    }
  }
  required_version = ">= 1.1"
}

provider "azurerm" {
  features {
    key_vault {
      purge_soft_delete_on_destroy    = true
      recover_soft_deleted_key_vaults = true
    }
  }
  subscription_id = ""
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v4.50.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5f3dcd2]

goroutine 546 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/managedredis.(*ManagedRedisDataSource).Read.ManagedRedisDataSource.Read.func1({0x8dcc1c0, 0xc0015b1730}, {0xc00254cd88, {0x8dcfc28, 0xc000043980}, 0xc005a19e60, 0x0, {0x8dcfe60, 0xe206440}})
github.com/hashicorp/terraform-provider-azurerm/internal/services/managedredis/managed_redis_data_source.go:258 +0x832
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*DataSourceWrapper).DataSource.func2({0x8dcc1c0, 0xc0015b1730}, 0x8db1f30?, {0x7429560?, 0xc00254cd88?})
github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_data_source.go:51 +0x145
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*DataSourceWrapper).DataSource.(*DataSourceWrapper).diagnosticsWrapper.diagnosticsWrapper.func3({0x8dcc1c0?, 0xc0015b1730?}, 0x1176592e000?, {0x7429560?, 0xc00254cd88?})
github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:218 +0x59
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000a30200, {0x8dcc118, 0xc002313e90}, 0xc005a19e60, {0x7429560, 0xc00254cd88})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:866 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000a30200, {0x8dcc118, 0xc002313e90}, 0x0?, {0x7429560, 0xc00254cd88})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:1088 +0x13e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000043de8, {0x8dcc118?, 0xc002313dd0?}, 0xc002313aa0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/grpc_provider.go:1889 +0x6ac
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadDataSource(0xc000c02d80, {0x8dcc118?, 0xc002313b00?}, 0xc002313aa0)
github.com/hashicorp/terraform-plugin-mux@v0.21.0/tf5muxserver/mux_server_ReadDataSource.go:36 +0x190
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc001a93ea0, {0x8dcc118?, 0xc0023132f0?}, 0xc001f51450)
github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/tf5server/server.go:736 +0x26d
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x83bdf40, 0xc001a93ea0}, {0x8dcc118, 0xc0023132f0}, 0xc005917800, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:843 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a2a200, {0x8dcc118, 0xc002313260}, 0xc001f13800, 0xc001835350, 0xe1d28d0, 0x0)
google.golang.org/grpc@v1.75.1/server.go:1431 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc000a2a200, {0x8dd01d8, 0xc0018f8000}, 0xc001f13800)
google.golang.org/grpc@v1.75.1/server.go:1842 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
google.golang.org/grpc@v1.75.1/server.go:1061 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 67
google.golang.org/grpc@v1.75.1/server.go:1072 +0x11d

Error: The terraform-provider-azurerm_v4.50.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

plan ok

### Actual Behaviour

plugin crash

### Steps to Reproduce

crash and other data source is err

### Important Factoids

_No response_

### References

maybe..

https://github.com/hashicorp/terraform-provider-azurerm/blob/v4.50.0/internal/services/managedredis/managed_redis_data_source.go#L250

```go
			if model := dbResp.Model; model != nil {
				if props := model.Properties; props != nil {
					defaultDb := DefaultDatabaseDataSourceModel{
						AccessKeysAuthenticationEnabled: strings.EqualFold(pointer.FromEnum(props.AccessKeysAuthentication), string(databases.AccessKeysAuthenticationEnabled)),
						ClientProtocol:                  pointer.FromEnum(props.ClientProtocol),
						ClusteringPolicy:                pointer.FromEnum(props.ClusteringPolicy),
						EvictionPolicy:                  pointer.FromEnum(props.EvictionPolicy),
						GeoReplicationGroupName:         flattenGeoReplicationGroupName(props.GeoReplication),
						GeoReplicationLinkedDatabaseIds: flattenLinkedDatabases(props.GeoReplication.LinkedDatabases),
						Module:                          flattenModules(props.Modules),
						Port:                            pointer.From(props.Port),
					}

					if props.GeoReplication != nil {
						defaultDb.GeoReplicationLinkedDatabaseIds = flattenLinkedDatabases(props.GeoReplication.LinkedDatabases)
					}

```

`if props.GeoReplication` is after 251 line?

--- a/internal/services/managedredis/managed_redis_data_source.go
+++ b/internal/services/managedredis/managed_redis_data_source.go
@@ -255,7 +255,6 @@ func (r ManagedRedisDataSource) Read() sdk.ResourceFunc {
 						ClusteringPolicy:                pointer.FromEnum(props.ClusteringPolicy),
 						EvictionPolicy:                  pointer.FromEnum(props.EvictionPolicy),
 						GeoReplicationGroupName:         flattenGeoReplicationGroupName(props.GeoReplication),
-						GeoReplicationLinkedDatabaseIds: flattenLinkedDatabases(props.GeoReplication.LinkedDatabases),
 						Module:                          flattenModules(props.Modules),
 						Port:                            pointer.From(props.Port),
 					}

